src/main/java/com/pharma/prescription/service/impl/DoctorServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package com.pharma.prescription.service.impl;
+
+import com.pharma.prescription.model.Doctor;
+import com.pharma.prescription.repository.IDoctorRepository;
+import com.pharma.prescription.service.IDoctorService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class DoctorServiceImpl implements IDoctorService {
+
+    private final IDoctorRepository doctorRepository;
+
+    @Autowired
+    public DoctorServiceImpl(IDoctorRepository doctorRepository){
+        this.doctorRepository = doctorRepository;
+    }
+
+    @Override
+    public Doctor findByEmail(String email) {
+        return doctorRepository.findByEmail(email);
+    }
```

Comment:
You need to implement DoctorNotFOUNDException

Reply:
Updated by adding the exception handling procedure